globus_automate_client/cli/flows.py

```diff
@@ -139,14 +168,24 @@ def flow_deploy(
         callback=flows_endpoint_envvar_callback,
     ),
     verbose: bool = verbosity_option,
+    input_format: FlowInputFormat = typer.Option(
+        FlowInputFormat.json,
+        "--input",
+        "-i",
+        help="Input format.",
+        case_sensitive=False,
+        show_default=True,
+    ),
 ):
     """
     Deploy a new Flow.
     """
     fc = create_flows_client(CLIENT_ID, flows_endpoint)
-    flow_dict = json.loads(definition)
+
+    flow_dict = _process_definition(definition, input_format)
+
     if input_schema is not None:
-        input_schema_dict = json.loads(input_schema)
+        input_schema_dict = yaml.safe_load(input_schema)
```

Comment:
This seems to assume that the input schema is yaml. Probably need to check the input format flag here.

I guess a single input format flag means one couldn't specify, for example, the definition in json and the schema in yaml (or vice versa), but I'm good with that.

Reply:
When I wrote this I was assuming that we would like to be able to accommodate the definition in json and the schema in yaml or vice versa. I can update these spots to be more specific and throw an error when the format does not match.

In this case, yaml can load json just fine, so it's not assuming it's just using the superset to parse the input_schema regardless of the format.